### ISSUE TITLE
Ongoing Transmission web PRs

### ISSUE THREAD

**Issue by Rukario** [transmission/transmission#7278]
### What is the issue?

@ckerr

S - Small PR (should be easy to review!)
\# - number as priority order I want to see merged sooner (up to 3)

Refactors
#7277
#7276
S #7001
3 S #6945

Features
2.1 #7263
#7257
#7051
#7008
#7000
1 #6977
#6948
S #6941
S #6867

Needs feedback/update
#7249 needs upcoming PR to tackle issues
#7053 needs split into multiple PRs @lisenenkov
#6985 depends on #7277
#6940 waiting for more important PRs to ship (#6948 and #6977)
#6737 superseded by #6977
2.2 #6681 depends on #7263 or new UI design
#6017 needs clarification

### Which application of Transmission?

Web client

### Which version of Transmission?

4.1.0-dev
